benchmark/run_all_benchmarks.sh

```diff
@@ -199,7 +221,7 @@ run_preconditioner_benchmarks() {
         for prec in ${PRECISIONS}; do
             echo -e "\t\t running jacobi ($prec) for block size ${bsize}" 1>&2
             cp "$1" "$1.imd" # make sure we're not loosing the original input
-            ./preconditioner/preconditioner \
+            ./preconditioner/preconditioner${BENCH_SUFFIX} \
```

Comment:
I guess this will not work for `float`. I suppose an exception will be thrown for that somewhere?

Reply:
Why would this not work for `float`? I use the suffix `_single` for these benchmarks (the suffixes are defined in the [CMakeLists.txt here](https://github.com/ginkgo-project/ginkgo/blob/change_benchmark_precision/benchmark/CMakeLists.txt#L92), and the translation in the [run_all_benchmarks.sh here](https://github.com/ginkgo-project/ginkgo/blob/change_benchmark_precision/benchmark/run_all_benchmarks.sh#L68))

Reply:
I mean because the block-Jacobi test seems to be going down two precisions in the array on https://github.com/ginkgo-project/ginkgo/pull/675/files#diff-def408061e3ffca8b3e80d667589f88c99f23d6fd482c0dde9b60579872f2938R211 I guess at least one of 0,2 or 2,0 implies you'll need a float8 if you start at float32? Correct me if I'm wrong please.

Reply:
Ah, you meant specifically for the various precision we try in this loop.
It will work without any exceptions since this is supported by the Jacobi preconditioner.
These numbers represent how many steps down we go in terms of precision. If we reach the lowest supported precision, we simply don't go further. That means in case of single precision (`Jacobi<float>`), `PRECISION=0,1` will lead to the same result as `PRECISION=0,2`, meaning they will both use half precision for storage. The `PRECISION` values will directly go to `precision_reduction`, which is detailed [in the Jacobi documentation](https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/preconditioner/jacobi.hpp#L418).

If we specify a lower precision than double precision, we do some redundant work, but I think that is fine, especially since we could add a lower precision than half (not likely, but possible).

Reply:
Okay, I see. Thanks.